backend/src/main/java/com/onlinebookstore/service/UserServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.onlinebookstore.service;
+
+import com.onlinebookstore.model.Role;
+import com.onlinebookstore.model.RoleName;
+import com.onlinebookstore.model.User;
+import com.onlinebookstore.repository.user.RoleRepository;
+import com.onlinebookstore.repository.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
+
+    @Override
+    public User register(User user)
+            throws IllegalArgumentException {
```

Comment:
```suggestion
    public User register(User user) {
```
why? it's RuntimeException